resource.go

```diff
@@ -43,6 +43,22 @@ func newResource(url string, r io.Reader) (*resource, error) {
 	}, nil
 }
 
+func newResourceJSON(url string, doc any) (*resource, error) {
```

Comment:
this method is not needed. change `newResource` to take `doc interface{}`

Reply:
applied